go/arrow/csv/reader.go

```diff
@@ -321,6 +324,10 @@ func (r *Reader) initFieldConverter(field *arrow.Field) func(array.Builder, stri
 				field.(*array.StringBuilder).Append(str)
 			}
 		}
+	case *arrow.TimestampType:
+		return func(field array.Builder, str string) {
+			r.parseTimestamp(field, str)
+		}
```

Comment:
rather than using a `WithTimestampUnit` option, just change this switch to be: `switch dt := field.Type.(type)` and then pass `dt.Unit` to the `parseTimestamp` function. We don't need the extra option since we *require* a schema currently.

The `WithTimestampUnit` option could be useful if we add functionality to infer the types from the columns rather than requiring a schema.

Reply:
@zeroshade make perfect sense, I knew I was missing something fundamental when adding that option.

I have update the PR with a commit to address the feedback.

Cheers for the review.